tests/rptest/tests/transactions_test.py

```diff
@@ -434,3 +444,132 @@ def get_tx_manager_broker():
+        by the transaction coordinator, and received at the consumer group and
+        topic leaders.
+        """
+        src_id = self.redpanda.idx(src_node)
+        dst_id = self.redpanda.idx(dst_node)
+        admin = self.redpanda._admin
+        # Have the transaction coordinator start out on 'src_node'.
+        wait_until(
+            lambda: admin.transfer_leadership_to(namespace="kafka_internal",
+                                                 topic="tx",
+                                                 partition=0,
+                                                 target_id=src_id),
+            timeout_sec=30,
+            backoff_sec=1)
+        # The receivers of transaction coordinator RPCs (the leader of the
+        # topic being written to and the group topic) should be on 'dst_node'.
+        wait_until(
+            lambda: admin.transfer_leadership_to(namespace="kafka",
+                                                 topic="__consumer_offsets",
+                                                 partition=0,
+                                                 target_id=dst_id),
+            timeout_sec=30,
+            backoff_sec=1)
+        wait_until(
+            lambda: admin.transfer_leadership_to(namespace="kafka",
+                                                 topic=self.output_t.name,
+                                                 partition=0,
+                                                 target_id=dst_id),
+            timeout_sec=30,
+            backoff_sec=1)
+
+        def run_txn(should_commit):
+            """
+            Runs through the steps of a transaction, using consumer groups. If
+            'should_commit' is True, commits the transactions, otherwise
+            aborts.
+
+            NOTE: this uses confluent_kafka, which doesn't do anything fancy in
+            the background, e.g. aggressively fetching coordinator leaders.
+            """
+            self.generate_data(self.output_t, self.max_records)
+            producer = ck.Producer({
+                'bootstrap.servers': self.redpanda.brokers(),
+                'transactional.id': str(self.txn_id),
+            })
+            producer.init_transactions()
+            consumer = ck.Consumer({
+                'bootstrap.servers': self.redpanda.brokers(),
+                'group.id': "test",
+                'auto.offset.reset': 'earliest',
+                'enable.auto.commit': False,
+            })
+
+            consumer.subscribe([self.output_t])
+            num_consumed_records = 0
+
+            producer.begin_transaction()
+            while num_consumed_records < self.max_records:
+                records = self.consume(consumer)
+                for r in records:
+                    producer.produce(self.output_t.name, r.value(), r.key())
+                num_consumed_records += len(records)
+
+            producer.send_offsets_to_transaction(
```

Comment:
This API does not supported old version. 
But also I think you do not needed it here